src/shared/bus-unit-util.c

```diff
@@ -2937,3 +2938,73 @@ int bus_service_manager_reload(sd_bus *bus) {
 
         return 0;
 }
+
+int unit_freezer_restore(const char *name, UnitFreezer *ret) {
+        _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+        _cleanup_free_ char *namedup = NULL;
+        int r;
+
+        assert(name);
+        assert(ret);
+        assert(!ret->name);
+        assert(!ret->bus);
```

Comment:
We usually don't do such checks. It's OK to assume the object would not be reused.